src/pages/homePage.jsx

```diff
@@ -42,16 +46,50 @@ function HomePage() {
       return;
     }
     const filteredList = moviesList.filter(movie => {
-      const formatedGenres = movie.genres.map(genre => genre.toUpperCase());
-      return formatedGenres.includes(filter.toUpperCase());
+      const formattedGenres = movie.genres.map(genre => genre.toUpperCase());
+      return formattedGenres.includes(filter.toUpperCase());
     });
     setMovies(filteredList);
   };
 
+  const deleteMovie = () => {
+    setMovies(movies.filter(movie => movie.id !== selectedMovie.id));
```

Comment:
Лучше разделять на этапы код:
1. Подготовка данных.
2. Обновление стейте.
Так же все функции сортировки, фильтрации и т.д. выносить в хелперы в файлы хелперов. Таким образом ты выделяешь функции, которые можно будет переиспользовать. В данном случае ты будешь неоднократно создавать похожие функции в разных местах и даже не будешь знать об дублировании
```suggestion
    const updatedMovieList = movies.filter(movie => movie.id !== selectedMovie.id);
    setMovies(updatedMovieList);
```